source/concurrent/Promise.ooc

```diff
@@ -19,7 +19,19 @@ _Synchronizer: abstract class {
 	_state: _PromiseState
 	init: func
 	wait: abstract func (time: TimeSpan) -> Bool
-	wait: func ~forever -> Bool { this wait(TimeSpan maximumValue) }
+	wait: func ~forever -> Bool {
+		version(debugDeadlock) {
+			timeLimit_msec := 2_000
+			timer := CpuTimer new() . start()
+			result := this wait(TimeSpan milliseconds(timeLimit_msec))
```

Comment:
> Alright, it's not very important but it would clean up the code. What about CpuTimer, why not use WallTimer ?

`WallTimer` could work, but since `_Task wait(TimeSpan)` uses a `CpuTimer`, a timeout might not trigger a deadlock then right?